mastodon/src/main/java/org/joinmastodon/android/model/Instance.java

```diff
@@ -154,6 +154,14 @@ public boolean isPixelfed() {
 		return version.contains("compatible; Pixelfed");
 	}
 
+	public boolean isIceshrimp() {
+		return version.contains("compatible; Iceshrimp");
+	}
+
+	public boolean isIceshrimpJs() {
+		return version.contains("compatible; Iceshrimp "); // Iceshrimp.NET will not have a space immediately after
+	}
```

Comment:
What are the difference between the two?

Reply:
`isIceshrimp` will match instances that use either Iceshrimp-JS (the Misskey fork) or Iceshrimp.NET (the rewrite). `isIceshrimpJs` only matches Iceshrimp-JS

Reply:
I meant in terms of supported/missing features. Is Iceshrimp.NET supposed to replace Iceshrimp-JS? In that case it might be better to only support the former, or add a comment to remove the latter later on.

Reply:
Iceshrimp.NET is supposed to replace Iceshrimp-JS, but that doesn't mean that every server will move to it. The differences aren't only in terms of supported Mastodon features. There are some features that work entirely differently. For instance, reactions and favorites are the same thing on Iceshrimp-JS, so favoriting a post means adding a reaction, but on Iceshrimp.NET they are two separate things (like how it works on Glitch forks and Akkoma). Therefore, the synchronization of favorites and reacts should only happen on Iceshrimp-JS (and in the future potentially also other servers that work in the same way, i.e. Sharkey)

I don't really see a problem with continuing to have code for both. Why remove a feature in the future if there might still be users who depend on it? I get that more code makes it harder to maintain, but the same code could be used for more servers so all that would be removed is a few if cases

Reply:
Ah ok, so Iceshrimp.NET will continue to exist separately and be updated? I don't think we should support outdated software in general, as it will inevitably break and/or have security vulnerabilities. I don't have a problem with supporting it until they do.

Reply:
The security of the server software really shouldn't be a concern to the client. These few conditionals that purely affect the UI don't have any impact on the security for any user of the client since the problems would be about the server software they use, which in this situation is unrelated to the client/this app. If someone has decided to use outdated and insecure server software, the client is not going to have an impact on that 

And again, assuming the code would be reused for users of other server-side software (which would be a good idea), all "continuing to support" means is to continue to have a couple of conditional cases that enable or disable UI elements/functions

Reply:
might also add that iceshrimp-js will not be outdated and will continue to receive security updates and bug fixes even as iceshrimp.net becomes the main focus

Reply:
I didn't know that it still would. Then there doesn't seem to be any issue here